### ISSUE TITLE
Synchronous `derived` stores should not trigger `require-stores-init`

### ISSUE THREAD

**Issue by mcous** [sveltejs/eslint-plugin-svelte#1073]
## Overview

I've been using the `3.0.0-next` version in our Svelte project, and I'm really liking the new recommended set of rules! However, I've had to disable `require-stores-init`, because it triggered on nearly every single usage of `derived`.

## Proposal

I think `require-stores-init` makes a lot of sense for `writable` and `readable` stores - it's goal seems to be to prevent an implicitly `undefined` store values. However, **I think synchronous `derived` stores should not trigger the `require-stores-init` lint**. If a `derived` store is synchronous, there is no risk that its value can be implicitly undefined, because it will run its callback synchronously when it gets its first subscriber.

If you agree with this change, I'm happy to contribute a PR!

## Alternatives

Alternatively, I could image a configuration option that allows you to selectively enable or disable `require-stores-init` for each store type. However, I think enforcing this rule on asynchronous `derived` stores is valuable, so I prefer my proposal above.

**Comment by baseballyama** [comment]
It might be good to add an option like `ignoreDerived` (default is false).

**Comment by mcous** [comment]
Alternatively, with stores on the way out with Svelte 5, maybe it's simply better to remove `require-stores-init` from the `recommended` config.

`ignoreDerived` could be a pragmatic solution, but I don't thinkg it's ideal for actual `derived` usage. I think an `ignoreDerived` option isn't the ideal solution, because derived stores may be asynchronous, and in that situation, I _do_ want to enforce an initial value:

```ts
const source = writable(0)

// synchronous, initial value not needed
const doubled = derived(source, ($source) => $source * 2)
 
// async., initial value needed
const doubledLater = derived(
  source,
  ($source, set) => {
    const id = setTimeout(() => set($source * 2), 1000)
    return () => clearTimeout(id)
  },
  0
)
```

**Comment by baseballyama** [comment]
I decided to remove it from the recommended config.👍
